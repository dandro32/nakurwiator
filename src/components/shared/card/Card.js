import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


const Card = ({card}) => {
  const imageSrc = process.env.PUBLIC_URL+card.imgSrc; // eslint-disable-line no-undef
  return (
    <div className="card"> 
      <img className="card-img-top" src={imageSrc} alt={card.titile} /> 
      <div className="card-body">
        <h5 className="card-title">{card.title}</h5>
        <p className="card-text">{card.desc}</p>
        <p className="text-center">
          <Link to={card.href} className="btn btn-primary">
            {card.hrefTitle}
          </Link>
        </p>
      </div>
    </div>
  );
};

Card.propTypes = {
  card: PropTypes.shape({
    title: PropTypes.string.isRequired,
    desc: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
    hrefTitle: PropTypes.string.isRequired,
    imgSrc: PropTypes.string.isRequired,
  })
};

export default Card;