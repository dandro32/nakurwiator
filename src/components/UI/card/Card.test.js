import React from 'react';
import Card from './Card';
import {MemoryRouter as Router} from 'react-router-dom';
import TestRenderer from 'react-test-renderer';

describe('Card component tests', ()=> {
  let result;
  const mockCard = {
    title: 'title',
    imgSrc: 'imgSrc',
    desc: 'desc',
    href: 'href',
    hrefTitle: 'hrefTitle'
  };
  beforeEach(() => {
    result = TestRenderer.create(
      <Router>
        <Card card={mockCard} />
      </Router>
      
    ).toJSON();

  });

  it('renders correctly', () => {
    expect(result).toMatchSnapshot();
  });
});