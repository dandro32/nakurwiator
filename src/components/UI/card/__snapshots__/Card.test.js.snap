// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component tests renders correctly 1`] = `
<div
  className="card"
>
  <img
    alt={undefined}
    className="card-img-top"
    src="imgSrc"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      title
    </h5>
    <p
      className="card-text"
    >
      desc
    </p>
    <p
      className="text-center"
    >
      <a
        className="btn btn-primary"
        href="/href"
        onClick={[Function]}
      >
        hrefTitle
      </a>
    </p>
  </div>
</div>
`;
